### DISCUSSION TITLE
Is it necessary to try running with Obfuscated JAR

### DISCUSSION THREAD

**Discussion by DelusionArun** [CastleLab/COMP3021-2021Fall-PA1-Student-Version#30] [Q&A] [answered]
I mostly checked my program using IntelliJ main configurations. Is it the same as running with jar or do we need to try checking with Obfuscated JAR as well for unexpected errors?

I get "Unable to access jarfile PA1-obf.jar" when I try running with  java --enable-preview -jar PA1-o bf.jar 

Also, I get "?" for Unicode elements even after I change editor to UTF-8. Does this mean that Unicode characters are not supported in my device/console?

**Comment by Derppening** [comment, marked as answer]
> Is it the same as running with jar or do we need to try checking with Obfuscated JAR as well for unexpected errors?

Running your program using the IntelliJ configuration is just running your own implementation. You should probably compare it to the obfuscated version.

> I get "Unable to access jarfile PA1-obf.jar" when I try running with java --enable-preview -jar PA1-o bf.jar

Make sure that your working directory is in `artifacts`.

**Reply by DelusionArun** [reply]
After using the command from artifacts, I now get-
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.

**Reply by Derppening** [reply]
Is there an exception stack trace or other error message I can refer to?

**Reply by DelusionArun** [reply]
This is what I get-
![error](https://user-images.githubusercontent.com/72679566/135447642-2d016914-08e7-4d61-845d-77d13c8e68a2.jpg)

The gradle configuration JAR compiles but I am not sure if anything happens. Should I edit the run configuration of JAR gradle instead?

**Reply by Derppening** [reply]
Check the output of `java -version`. I suspect the default Java version of your machine is Java 8.

**Reply by DelusionArun** [reply]
It appears that my java version states java version "10.0.2" even if my project structure and modules both state jdk 16

**Reply by Derppening** [reply]
Because your terminal does not care about what you set in IntelliJ or Gradle. The settings you put in IntelliJ only apply to IntelliJ itself. 

If you are using the terminal, you will need to explicitly specify the full path to your JDK 16 installation, e.g. `/usr/lib/jvm/java-16-openjdk/bin/java -jar ...`

**Comment by DelusionArun** [comment]
I seem to still have a new error after updating java- 
![error](https://user-images.githubusercontent.com/72679566/135484091-e3018b49-d436-47d7-91e0-4c16deb64dd3.jpg)

**Reply by Derppening** [reply]
Read the error message from the `FileNotFoundException`. Does that file exist?

What you instead want is, since the `puzzles` directory reside in the *parent* of the `artifacts` directory, `java --enable-preview -jar PA1-obf.jar ../puzzles/00-unicode-test.game`.

Note the `..`, which indicates you want to go up one level in the directory tree.

**Reply by DelusionArun** [reply]
Thank you very much. Now it works

**Comment by DelusionArun** [comment]
I still get "?" for Unicode elements even after I change editor to UTF-8. Does this mean that Unicode characters are not supported in my device/console?

**Reply by Derppening** [reply]
Yes, it is likely the case.

Just run the game without the `--unicode` flag.

**Reply by DelusionArun** [reply]
Thank you again
